### PR TITLE
Sort DAGs

### LINKED ISSUE #19138: Prioritize larger test spec files first
## Reason
Sometimes, the TestHarness will run a large DAG near the end. This is will cause excessive overall run times on Civet, due to the --pedantic-check nature of running each test in a test spec file serially. One second at a time.

I.e, the misc/check_error test spec file with 115 tests. If this test were to launch last, it add 115 seconds at minimum to the overall testing suite.

## Design
Prioritize larger DAG objects to launch before others. This means we need to create all tester objects and scrutinize them before launching.

## Impact
Bring the average testing time down on Civet

### PR DESCRIPTION
**Pull request by milljm**
Sort the DAG objects by how large they are, and launch them.
Remove unnecessary sets/locks/lists.

Closes #19138

### CODE REVIEW

**Comment by milljm** [comment]
Rebased on devel.

File: python/TestHarness/schedulers/Scheduler.py [hunk 1]
```diff
@@ -154,41 +143,43 @@ def notifyFinishedSchedulers(self):
...
 
+    def retrieveJobs(self):
+        """ return all the jobs the scheduler was tasked to perform work for """
+        return [x[1].topological_sort() for x in self.__dag_bank]
+
+    def __sortAndLaunch(self):
+        """
+        Sort by largest DAG and launch
+        """
+        sorted_jobs = sorted(self.__dag_bank, key=lambda x: len(x[1].topological_sort()), reverse=True)
```
[the remark this answers is not in the document]
Comment (dschwen) [reply python/TestHarness/schedulers/Scheduler.py:163]: So you're running the biggest jobs (meaning `tests` specs with the most entries) first?

[the remark this answers is not in the document]
Comment (milljm) [reply python/TestHarness/schedulers/Scheduler.py:163]: Correct. It solves the odd event were misc/error_check runs near last. Adding over 100 seconds to the --pedantic-tests run time (as described in the linked issue #19138).

Looks like I have some new failures in unittests though. So this PR will need an update.

File: python/TestHarness/schedulers/Scheduler.py [hunk 2]
```diff
@@ -99,20 +99,13 @@ def __init__(self, harness, params):
         # Private set of jobs currently running
         self.__active_jobs = set([])
 
-        # Jobs that are taking longer to finish than the alloted time are reported back early to inform
-        # the user 'stuff' is still running. Jobs entering this set will not be reported again.
+        # jobs that are taking longer to finish than the alloted time are reported back early to inform
+        # the user 'stuff' is still running. jobs entering this set will not be reported again.
```
Comment (cticenhour) [inline python/TestHarness/schedulers/Scheduler.py:103]:
```suggestion
        # Jobs that are taking longer to finish than the alloted time are reported back early to inform
        # the user 'stuff' is still running. Jobs entering this set will not be reported again.
```

Victim of find and replace?

**Review by loganharbour — APPROVED** [review]

File: python/TestHarness/schedulers/Scheduler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (milljm) [reply python/TestHarness/schedulers/Scheduler.py:166]: resolved

**Pull request merged**